decoder/src/frame.rs

```diff
@@ -156,9 +163,16 @@ fn format_args_real(
         buf: &mut String,
     ) -> Result<(), fmt::Error> {
         match hint {
-            Some(DisplayHint::Binary) => write!(buf, "{:#b}", x)?,
-            Some(DisplayHint::Hexadecimal { uppercase: false }) => write!(buf, "{:#x}", x)?,
-            Some(DisplayHint::Hexadecimal { uppercase: true }) => write!(buf, "{:#X}", x)?,
+            Some(DisplayHint::NoHint { zero_pad }) => write!(buf, "{:#01$}", x, zero_pad)?,
+            Some(DisplayHint::Binary { zero_pad }) => write!(buf, "{:#01$b}", x, zero_pad)?,
+            Some(DisplayHint::Hexadecimal {
+                uppercase: false,
+                zero_pad,
+            }) => write!(buf, "{:#01$x}", x, zero_pad)?,
+            Some(DisplayHint::Hexadecimal {
+                uppercase: true,
+                zero_pad,
+            }) => write!(buf, "{:#01$X}", x, zero_pad)?,
```

Comment:
I am slightly in favor of this pattern, cause I think it is more clear:
```suggestion
            Some(DisplayHint::Hexadecimal {
                uppercase,
                zero_pad,
            }) => match uppercase {
                false => write!(buf, "{:#01$X}", x, zero_pad)?,
                true => write!(buf, "{:#01$x}", x, zero_pad)?,
            },
```
_(same for `fn format_u128`)_